### PR TITLE
ui: Allow searching in dropdowns

### PR DESCRIPTION
**Pull request by davidjumani**
### Description

Allows searching in dropdowns
Part of #3121 separated for simplicity

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

### CODE REVIEW

**Comment by davidjumani** [comment]
@blueorangutan ui

File: ui/src/components/header/SamlDomainSwitcher.vue [hunk 1]
```diff
@@ -22,6 +22,11 @@
       :loading="loading"
       :defaultValue="currentAccount"
       :value="currentAccount"
+      showSearch
+      optionFilterProp="children"
+      :filterOption="(input, option) => {
+        return option.componentOptions.children[0].text.toLowerCase().indexOf(input.toLowerCase()) >= 0
+      }"
```
Comment (shwstppr) [inline ui/src/components/header/SamlDomainSwitcher.vue:29]: @davidjumani this code is common across all select items. Is it possible to extract this and have a custom select component?

**Comment by utchoang** [comment]
@davidjumani can you add a search in dropdown for the following sections?
`src\components\view\DedicateDomain.vue` line 41, 
`src\components\view\Login.vue` line 40, 107
`src\views\dashboard\CapacityDashboard.vue` line 23

File: ui/src/components/header/SamlDomainSwitcher.vue (hunk 1 above)
Comment (davidjumani) [reply ui/src/components/header/SamlDomainSwitcher.vue:29]: I think that might involve replicating every prop used which could potentially break things. @utchoang ?

**Comment by davidjumani** [comment]
@blueorangutan ui

**Comment by davidjumani** [comment]
@blueorangutan ui

**Comment by nvazquez** [comment]
@davidjumani can you please resolve the conflicts?

**Review by utchoang — APPROVED** [review]

**Comment by davidjumani** [comment]
@blueorangutan ui

**Comment by rhtyd** [comment]
@blueorangutan ui

**Comment by davidjumani** [comment]
@blueorangutan ui

**Review by nvazquez — APPROVED** [review]
LGTM

Minor enhancement request: is it possible to highlight the matched input text in the filtered results?
Screenshot from 2021-09-08 12-07-44

**Comment by davidjumani** [comment]
@nvazquez Don't think that might be possible as we use ant internal methods to filter the text cc @utchoang

**Comment by davidjumani** [comment]
@blueorangutan ui

**Comment by rhtyd** [comment]
@shwstppr are you lgtm in merge this, then changes are large - should they be manually tested?

**Review by shwstppr — APPROVED** [review]
LGTM. I've not tested each and every view but nature of the change shouldn't break anything

**Comment by nvazquez** [comment]
Agree with @shwstppr, merging based on manual tests performed by him and myself

**Pull request merged**